activesupport/test/core_ext/module/attribute_accessor_per_thread_test.rb

```diff
@@ -2,53 +2,56 @@
 require 'active_support/core_ext/module/attribute_accessors_per_thread'
 
 class ModuleAttributeAccessorPerThreadTest < ActiveSupport::TestCase
-  def setup
-    @class = Class.new do
-      thread_mattr_accessor :foo
-      thread_mattr_accessor :bar,  instance_writer: false
-      thread_mattr_reader   :shaq, instance_reader: false
-      thread_mattr_accessor :camp, instance_accessor: false
-    end
+  class Foo
```

Comment:
In this case I think it'd be better for it to raise: the previous behaviour treated all anonymous classes as one, so that doesn't seem terribly useful.